tokio-stream/src/wrappers/stream_close.rs

```diff
@@ -0,0 +1,59 @@
+use crate::Stream;
+use pin_project_lite::pin_project;
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+pin_project! {
+    /// Stream that returns None when the stream is exhausted. To be used with StreamMap to know when a stream ends.
+    #[must_use = "streams do nothing unless polled"]
+    pub struct StreamNotifyClose<S> {
```

Comment:
The name is not great — notify normally means something different. But I can't think of anything better.

Reply:
Totally agree. `StreamClose` implies nothing really.. would love suggestions.